app.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+require 'sinatra/base'
+require 'json'
+require 'haversine'
+require 'bigdecimal'
+require_relative 'lib/person_dto'
+require_relative 'lib/data_handlers/json_handler'
+require_relative 'lib/exceptions/data_unavailable_error'
+
+# Main entry point for routes
+class App < Sinatra::Base
+  
+  #sanitise user input
+  before do
+    params.each do |p, v|
+      params[p] = Rack::Utils.escape_html(v)
+    end
+  end
+  
+  # display all
+  get '/' do
+    # set defaults for bristol
+    @form_values = {params['calcAvgValue'] => 'off',
+                     'lat' => '51.450167', 'lon' => '-2.594678'}
+
+    json_handler = JsonHandler.new
+    begin
+      @people_stats = json_handler.get_person_dtos(true)
+    rescue DataUnavailableError => error
+      halt 400, error.message
+    end
+    status 200
+    erb :index
+  end
+
+  # filter and calculate average value depending on user input
+  post '/' do
+    @people_stats = {}
+    @form_values = {'lon'          => params['lon'],
```

Comment:
I would load params into a request model class. e.g.  `CalculateRequestModel` with `attr_accessors` for each param to avoid pulling data from the params hash by string key everywhere in code.